### DISCUSSION TITLE
Get-PnPPage produces reproducible error on certain pages

### DISCUSSION THREAD

**Discussion by matthias-glubrecht** [pnp/powershell#1442] [Q&A]
I have a modern page (Article page) in SharePoint online that causes an error when I try to get it with Get-PnPPage.
The error message is
Get-PnPPage : 'S' is invalid after a value. Expected either ',', '}', or ']'. Path: $ | LineNumber: 0 | BytePositionInLine: 1748.
In Zeile:1 Zeichen:1
+ Get-PnPPage "vitero.aspx" -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Get-PnPPage], JsonException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Pages.GetPage

I can supply the value of CanvasContent1 if it helps in debugging. It's in the attached file contents.txt.
[contents.txt](https://github.com/pnp/powershell/files/7705541/contents.txt)

**Comment by raytri** [comment]
Was this ever resolved? It's almost two years later, and I'm running into the exact same issue.

I haven't noticed an obvious rhyme or reason to the issue. Many of the pages have had their title changed after creation, but that's very common in our organization. And not all pages where the title and FileLeafRef are obviously different have problems.

An example of the problem I'm having:

**Page title:** "AD&D insurance, Local 12"
**FileLeafRef** "Benefits-ADD-insurance-GHI-Local-12.aspx"

**Code**:
$pageDetails = Get-PnPPage -Identity "Benefits-ADD-insurance-GHI-Local-12.aspx"

**Error**:
Get-PnPPage: 's' is invalid after a value. Expected either ',', '}', or ']'. Path: $ | LineNumber: 0 | BytePositionInLine: 13171.

**Comment by matthias-glubrecht** [comment]
I don‘t think it was resolved. I worked around it, by using the REST-API directly for what I was trying to achieve. Unfortunately I didn‘t have the time to look into the source code to analyze it further.

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: raytri ***@***.***>
Gesendet: Friday, August 11, 2023 8:27:06 PM
An: pnp/powershell ***@***.***>
Cc: Matthias Glubrecht ***@***.***>; Author ***@***.***>
Betreff: Re: [pnp/powershell] Get-PnPPage produces reproducible error on certain pages (Discussion #1442)


Was this ever resolved? It's almost two years later, and I'm running into the exact same issue.

I haven't noticed an obvious rhyme or reason to the issue. Many of the pages have had their title changed after creation, but that's very common in our organization. And not all pages where the title and FileLeafRef are obviously different have problems.

An example of the problem I'm having:

Page title: "AD&D insurance, Local 12"
FileLeafRef "Benefits-ADD-insurance-GHI-Local-12.aspx"

Code:
$pageDetails = Get-PnPPage -Identity "Benefits-ADD-insurance-GHI-Local-12.aspx"

Error:
Get-PnPPage: 's' is invalid after a value. Expected either ',', '}', or ']'. Path: $ | LineNumber: 0 | BytePositionInLine: 13171.

**Comment by tom-daly** [comment]
This issue is happening for me as well only on 1 page right now

![image](https://github.com/pnp/powershell/assets/9259305/76eae628-915a-44f6-9424-851f2a70b274)

**Reply by tom-daly** [reply]
PnP.PowerShell 2.2.57

**Comment by simoncumming** [comment]
I am experiencing the same issue as well with a page and I feel like the reason may be that it used to have a forward slash in it but now does not

![image](https://github.com/pnp/powershell/assets/19344069/0fa4bb9d-7d89-40f4-873f-b3c33166aa23)

![image](https://github.com/pnp/powershell/assets/19344069/89ff9eac-e57c-41c4-8cc2-484a7bec94cc)

Does anyone know if this is something suitable to be logged as an issue in this project? Or is it part of the PnP Framework?

**Reply by simoncumming** [reply]
I realised what was causing the issue in my case.

I had a webpart on the page that doesn't seem to escape the " characters in its PropertiesJSON properly. I removed the " from the content and no longer received the error message.
